addons/stock/tests/test_warehouse.py

```diff
@@ -929,3 +929,51 @@ def test_modified_global_route(self):
         new_mto_route = route_sudo.search([("name", "=", "Replenish on Order (MTO)")])
         self.assertEqual(len(new_mto_route), 1)
         self.assertEqual(new_mto_route.company_id.id, company_2.id)
+
+    def test_search_qty_to_order_of_0(self):
+        """
+        Test that searching for orderpoints with `qty_to_order = 0` works correctly.
+
+        The test first creates two orderpoints with `qty_to_order_manual = 0`,
+        then updates `qty_to_order` to a non-zero value to ensure the search
+        properly filters records based on the field value.
+        """
+        warehouse_A = self.env['stock.warehouse'].create({
+            'name': 'Warehouse A',
+            'code': 'WH_A',
+            'delivery_steps': 'pick_pack_ship',
+        })
+
+        # qty_to_order_computed = 0, qty_to_order_manual = 0
```

Comment:
```suggestion
```
Comments are often useless
Here the code is clear enough
(same for all next comments)